lib/style-property-matcher.js

```diff
@@ -24,41 +24,67 @@ var StylePropertyMatcher = (function (window, document) { // jshint ignore:line
+     * Other browsers encloses such value with double quotes.
+     * In order to normalize this, we do a quick check here.
+     */
+    var quote, useFallback = false;
+
+    (function() {
+        var root = document.createElement('div');
+        var styleEl = document.createElement('style');
+        styleEl.appendChild(document.createTextNode('#EXTCSS_TEMP::after{content:""}'));
+        var testEl = document.createElement('div');
+        testEl.setAttribute('style', 'display:none!important;');
+        testEl.id = "EXTCSS_TEMP";
+        root.appendChild(styleEl);
+        root.appendChild(testEl);
+        document.body.appendChild(root);
+        var propValue = window.getComputedStyle(testEl, ':after').getPropertyValue('content');
+        if(propValue.length === 0 && typeof window.getMatchedCSSRules == 'function') {
+            useFallback = true;
+            propValue = window.getMatchedCSSRules(testEl, ':after')[0].style.getPropertyValue('content');
+        }
+        quote = propValue.charAt(0);
+        document.body.removeChild(root);
+    })();
+
+
     /**
      * There is a known issue in Safari browser:
      * getComputedStyle(el, ":before") is empty if element is not visible.
      * 
      * To circumvent this issue we use getMatchedCSSRules instead.
+     *
+     * It appears that getMatchedCSSRules sorts the CSS rules
+     * in the increasing order of the specifity of corresponding selectors.
      */
-    var getPseudoElementComputedStyle = function(element, pseudoElement) {
-        var styles = [];
-        var cssRules = window.getMatchedCSSRules(element, pseudoElement) || [];
-
-        var iCssRules = cssRules.length;
-        while (iCssRules--) {
-            var style = cssRules[iCssRules].style;
-            var iStyle = style.length;
-            while (iStyle--) {
-                var name = style[iStyle];
-                styles[name] = style.getPropertyValue(name);
-                styles.push(name);
-            }
+    function CSSStyleDec(cssRules) {
+        this.cssRules = cssRules;
+    }
+
+    CSSStyleDec.prototype.getPropertyValue = function(name) {
+        var i = this.cssRules.length, value;
+        while(i-- > 0 && !value) {
+            value = this.cssRules[i].style.getPropertyValue(name);
         }
+        return value;
+    };
 
-        styles.sort();
-        styles.getPropertyValue = function(name) {
-            return styles[name];
-        };
-        return styles;
+    var getPseudoElementComputedStyle = function(element, pseudoElement) {
+        return new CSSStyleDec(window.getMatchedCSSRules(element, pseudoElement) || []);
     };
 
     /**
      * Unquotes specified value
      */
     var removeDoubleQuotes = function(value) {
-        if (typeof value === "string" && value.length > 1 && value[0] === '"' && value[value.length - 1] === '"') {
-            // Remove double-quotes
-            value = value.substring(1, value.length - 1);
+        var l;
```

Comment:
Ok. Sometimes it's hard to understand constructions like this while code reviewing. If you want to do like this, please write it clearly)